plugins/http_plugin/http_plugin.cpp

```diff
@@ -59,6 +58,28 @@ namespace eosio {
       return api_category::unknown;
    }
 
+   [[gnu::const]] 
+   std::string category_plugin_name(api_category category) {
+      if (category == api_category::db_size)
+         return "eosio::db_size_api_plugin";
+      if (category == api_category::trace_api)
+         return "eosio::trace_api_plugin";
+      if (category == api_category::prometheus)
+         return "eosio::prometheus_plugin";
+      if (category == api_category::test_control)
+         return "eosio::test_control_plugin";
+      if (api_category_set({api_category::chain_ro, api_category::chain_rw}).contains(category))
+         return "eosio::chain_api_plugin";
+      if (api_category_set({api_category::net_ro, api_category::net_rw}).contains(category))
+         return "eosio::net_api_plugin";
+      if (api_category_set({api_category::producer_ro, api_category::producer_rw, api_category::snapshot})
+              .contains(category))
+         return "eosio::producer_api_plugin";
+      // It's a programming error when the control flow reaches this point, 
+      // please make sure all the plugin names are returned from above statements.
+      throw std::runtime_error("No correspding plugin for the category value");
```

Comment:
Seems like this should be `std::logic_error` or `FC_ASSERT` or `EOS_ASSERT` or even `assert()`

Reply:
I like `assert()` for things that logically should never happen. No point in adding code in release build for that, and a `throw` can be caught, leaving the programmer unaware that his logic is wrong.